util/util.go

```diff
@@ -99,6 +104,31 @@ func GetMachinesForCluster(ctx context.Context, c client.Client, cluster *cluste
 	return &machines, nil
 }
 
+// Internal helper: returns normalized build version (with "v" prefix if needed)
+// If input doesn't match known version pattern, returns empty string.
+// Taken from k8s.io/cmd/kubeadm/app/util
+func NormalizedBuildVersion(version string) string {
+	if kubeReleaseRegex.MatchString(version) {
+		if strings.HasPrefix(version, "v") {
+			return version
+		}
+		return "v" + version
+	}
+	return ""
+}
+
+// KubernetesVersionToImageTag is helper function that replaces all
+// non-allowed symbols in tag strings with underscores.
+// Image tag can only contain lowercase and uppercase letters, digits,
+// underscores, periods and dashes.
+// Current usage is for CI images where all of symbols except '+' are valid,
+// but function is for generic usage where input can't be always pre-validated.
+// Taken from k8s.io/cmd/kubeadm/app/util
+func KubernetesVersionToImageTag(version string) string {
+	allowed := regexp.MustCompile(`[^-a-zA-Z0-9_\.]`)
```

Comment:
This can be also a global variable like the other one

Reply:
@sedefsavas We might want to add some validation in place for fields with `imageTag` to be compliant with the above regex, we could use `kubebuilder:validation:Pattern` https://book.kubebuilder.io/reference/markers/crd-validation.html

Reply:
image tags are in the kubeadm types, so we can't do much here atm. have added a comment in the webhook.